packages/marshal/src/marshal.js

```diff
@@ -285,59 +267,156 @@ function isPassByCopyRecord(val) {
   return true;
 }
 
-const makeRemotableProto = (oldProto, allegedName) => {
+// Below we have a series of paired predicate functions and assertion
+// functions. The semantics of the assertion function is just to assert that
+// the corresponding predicate function would have returned true. But it
+// reproduces the internal tests so failures can give a better error message.
+// Please maintain these pairs so their internal logic means the same thing,
+// and so that they read as parallel to each other as possible.
+
+/**
+ * @param {*} iface
+ * @returns {boolean}
+ */
+const isIface = iface =>
+  typeof iface === 'string' &&
+  (iface === 'Remotable' || iface.startsWith('Alleged: '));
+
+/**
+ * @param {InterfaceSpec} iface
+ */
+const assertIface = iface => {
+  // TODO other possible ifaces, once we have third party veracity
+  assert.typeof(
+    iface,
+    'string',
+    X`Interface ${iface} must be a string; unimplemented`,
+  );
+  assert(
+    iface === 'Remotable' || iface.startsWith('Alleged: '),
+    X`For now, iface ${q(
+      iface,
+    )} must be "Remotable" or begin with "Alleged: "; unimplemented`,
+  );
+};
+
+/**
+ * @param {Object|null} oldProto
+ * @param {InterfaceSpec} iface
+ * @returns {Object}
+ */
+const makeRemotableProto = (oldProto, iface) => {
   assert(
     oldProto === objectPrototype || oldProto === null,
     X`For now, remotables cannot inherit from anything unusual`,
   );
   // Assign the arrow function to a variable to set its .name.
-  const toString = () => `[${allegedName}]`;
+  const toString = () => `[${iface}]`;
   return harden({
     __proto__: oldProto,
     [PASS_STYLE]: REMOTE_STYLE,
     toString,
-    [Symbol.toStringTag]: allegedName,
+    [Symbol.toStringTag]: iface,
   });
 };
 
+/**
+ * @param {*} val
+ * @returns {boolean}
+ */
+const isRemotableProto = val => {
+  if (
+    !(typeof val === 'object' && !Array.isArray(val) && val !== null) &&
+    isFrozen(val)
```

Comment:
I may be wrong; it looks like this line returns false if the object was frozen, but the assertRemotableProto fails if the object was *not* frozen.

Let me know when you've merged these checks so that we are guaranteed consistency, then I'll review again.

Reply:
You're right. I got this test backward. It was only because I put the closing parens in the wrong place. Nothing intentional.

Reply:
Fixed.